pkg/intermediate/aggregate.go

```diff
@@ -648,6 +712,132 @@ func (a *AggregationProcess) addFieldsForStatsAggregation(record entities.Record
 	return nil
 }
 
+func (a *AggregationProcess) addFieldsForThroughputCalculation(record entities.Record, fillSrcStats, fillDstStats bool) error {
+	if a.aggregateElements == nil {
+		return nil
+	}
+	antreaFlowEndSecondsElements := a.aggregateElements.AntreaFlowEndSecondsElements
+	antreaThroughputElements := a.aggregateElements.ThroughputElements
+	antreaSourceThroughputElements := a.aggregateElements.SourceThroughputElements
+	antreaDestinationThroughputElements := a.aggregateElements.DestinationThroughputElements
+
+	var timeStart, timeEnd uint32
+	var byteCount, reverseByteCount uint64
+	timeStart, err := getUnsigned32ValueByIeName(record, "flowStartSeconds")
+	if err != nil {
+		return err
+	}
+	timeEnd, err = getUnsigned32ValueByIeName(record, "flowEndSeconds")
+	if err != nil {
+		return err
+	}
+	byteCount, err = getUnsigned64ValueByIeName(record, "octetTotalCount")
+	if err != nil {
+		return err
+	}
+	reverseByteCount, err = getUnsigned64ValueByIeName(record, "reverseOctetTotalCount")
+	if err != nil {
+		return err
+	}
+
+	// Initialize flowEndSecondsFromSourceNode and flowEndSecondsFromDestinationNode.
+	for _, ieName := range antreaFlowEndSecondsElements {
```

Comment:
Change: Adding a group `antreaFlowEndSecondsElements ` instead of "hard-coding" the fields name. Also to make the pattern more consistent with other extra fields at flowaggregator.go on Antrea side - [code](https://github.com/antrea-io/antrea/blob/86c414e3a566538ffd8d6ca01f0e84ca25031498/pkg/flowaggregator/flowaggregator.go#L124)